Charlie/v6 193 style wordpress core blocks as needed in responsive foundation

gcharlie opened the pull request:
This branch address inadequate and broken styles for several core blocks observed in https://github.com/bu-ist/id-gutenberg/issues/193

All the styles in this branch are intended to address layout concerns only (no opinionated styles)

### Sandbox Link

- **[Quote/Pullquote Block](https://id-charlie-upgrade-58.cms-devl.bu.edu/core-3/html-elements/additional-block-editor-page-example/):** The Standard Quote Block block was actually fixed with content area styles applied in https://github.com/bu-ist/responsive-foundation/pull/296. But the Pullquote Block need redundant margins eliminated. It's padding now uses CSS variables that can be easily overridden by child themes, additionally the inline padding uses `--bu--content--margin-inline` as a placeholder to be inline with other spacings by default.
- **[Separator Block](https://id-charlie-upgrade-58.cms-devl.bu.edu/core-3/html-elements/additional-block-editor-page-example/2/):** The separator block would unexpectedly grow when having a color applied to it, this has been eliminated and the vertically spacing on all 3 styles are made equal for consistency.
- **[Search Block](https://id-charlie-upgrade-58.cms-devl.bu.edu/core-3/html-elements/additional-block-editor-page-example/3/):** some redundant margins where elimited for the form elements in this block.
- **[Media and Text Block](https://id-charlie-upgrade-58.cms-devl.bu.edu/core-3/html-elements/additional-block-editor-page-example/4/):** I made the choice to zero out `--bu--content--padding-block` and `--bu--content--padding-inline` for this block when a background is applied, because this block already has its own means of creating spacing for the text when  a background is applied. Additionally I re enabled the font-size editor property, because by default the text for this block is set to have a custom font size, which can not be changed without this setting enabled.
- **[Query Loop](https://id-charlie-upgrade-58.cms-devl.bu.edu/core-3/html-elements/additional-block-editor-page-example/5/):** This Block is composed of several child blocks, to of which needed additional styling
  - **Post Template Block:** The styles added for this block first address redundant spacing and standardize margins and gaps, and secondly change the way the columns work in order to make them media query and container agnostic.
  - **Post Featured Image Block:** The styles added for this block primarily standardize the size of the featured image thru the use of the `aspect-ratio` property, by introducing a css variable and setting it to a fallback of 16:9.
- **[Table Block](https://id-charlie-upgrade-58.cms-devl.bu.edu/core-3/html-elements/additional-block-editor-page-example/6/):** ~~This removes some of the opinionated  responsive framework styles that conflict with and break some of the editor settings for this block.~~ Additional styling accommodations to create parity with the established responsive foundation table styles and core table block will preserving nesting color treatment and block color settings functionality


### Companion PR
- https://github.com/bu-ist/responsive-child-starter-3x-block-editor/pull/43

### Related Issues
- https://github.com/bu-ist/id-gutenberg/issues/193

gcharlie committed:
remove table colors that conflic with block editor

gcharlie committed:
remove legacy table margin

gcharlie committed:
fix padding usage

gcharlie committed:
zero out has-backgorund padding on media text block

gcharlie committed:
cleanup

gcharlie committed:
seperator style accomidations

gcharlie committed:
pullquote styles

gcharlie committed:
additional pullquote style refinement

gcharlie committed:
search block  styles

gcharlie committed:
initial pass at query loop block

gcharlie committed:
additional styles

gcharlie committed:
query loop column styles

gcharlie committed:
Merge branch 'v6' of https://github.com/bu-ist/responsive-foundation into charlie/v6-193-Style-Wordpress-Core-blocks-as-needed-in-Responsive-Foundation

gcharlie committed:
add basic comment text

gcharlie committed:
normalize featured image size

gcharlie committed:
set varibable with placeholder

gcharlie committed:
cleanup

acketon commented on burf-theme/content/blocks/_post-template.scss:
```diff
@@ -0,0 +1,99 @@
+	}
+}
+
+.wp-block-post-featured-image {
```
this class isn't nested so could it instead live in the `_post-featured-image.scss` partial?

acketon commented on burf-theme/content/blocks/_post-template.scss:
```diff
@@ -0,0 +1,99 @@
+	}
+}
+
+.wp-block-post-date {
```
similar for this class for the date block

acketon commented on burf-base/_tables.scss:
```diff
@@ -156,8 +102,6 @@ table {
 	}
 
 	thead {
-		background: $table-head-bg;
```
I'm hesitant to remove some of these for non-block tables at least. Could we add custom properties here? Maybe Crystal's PR is already doing that?

acketon reviewed (approved):
This looks good for the majority of it. I'd like to see block tables vs non-block tables side by side before deciding if it's okay to remove the existing caption and heading color styles from Foundation

gcharlie committed:
styles for button block

gcharlie committed:
parent block inheritense

gcharlie committed:
use bu variable for link color

gcharlie committed:
reset table styles

gcharlie committed:
table and table block homogenization

gcharlie committed:
table block styles

gcharlie committed:
remove spacing tesst colors

gcharlie committed:
clean up query loop styles

gcharlie committed:
tweak header contrast

gcharlie committed:
clean up table code

gcharlie commented:
- modified styles to accommodate deep block nesting.
  - https://id-charlie-upgrade-58.cms-devl.bu.edu/core-3/html-elements/lorem-ipsum-dolor-sit-amet/
- Additional styling accommodations to create parity with the established responsive foundation table styles and core table block will preserving nesting color treatment and block color settings functionality
- moved query loop image and date block styles into their own partials
- Styles for cure button block fixed: https://id-charlie-upgrade-58.cms-devl.bu.edu/core-3/html-elements/additional-block-editor-page-example/7/

gcharlie committed:
Merge branch 'v6' of https://github.com/bu-ist/responsive-foundation into charlie/v6-193-Style-Wordpress-Core-blocks-as-needed-in-Responsive-Foundation

acketon reviewed (approved):
Looks great
